File: 백준/Silver/11399. ATM/ATM.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      

        int N = Integer.parseInt(br.readLine());
        Integer[] num_list = new Integer[N];
        int sum=0;
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0;i<N;i++){
            int num = Integer.parseInt(st.nextToken());
            num_list[i]=num;
        }
        Arrays.sort(num_list);
        for(int i=1; i<N; i++){
            num_list[i]=num_list[i-1]+num_list[i];
        }
        for(int j=0;j<N;j++){
            sum+= num_list[j];
        }
        System.out.println(sum);
    }
    
}

File: 백준/Silver/11399. ATM/ATM.py
from sys import stdin
input = stdin.readline
# 인출하는 시간이 짧은 순으로 배열을 정렬하고 누적합 구하기

N =int(input())
time = list(map(int,input().split()))

time.sort()
#[1,2,3,3,4]
#[1,3,6,9,13]
for i in range(N-1):
    time[i+1]+=time[i]
print(sum(time))